src/pages/UserSettings/Teams.vue

```diff
@@ -0,0 +1,171 @@
+<script>
+import { mapActions, mapGetters } from 'vuex'
+import { clearCache } from '@/vue-apollo'
+
+import ManagementLayout from '@/layouts/ManagementLayout.vue'
+import ConfirmDialog from '@/components/ConfirmDialog'
+
+export default {
+  components: {
+    ManagementLayout,
+    ConfirmDialog
+  },
+  data() {
+    return {
+      dialogRemoveUser: false,
+      headers: [
+        { text: 'Name', value: 'name' },
+        { text: 'URL', value: 'slug' },
+        { text: 'Role', value: 'role' },
+        { text: 'Actions', value: 'id' }
+      ],
+      isRemovingUser: false,
+      removeTenant: null
+    }
+  },
+  computed: {
+    ...mapGetters('user', ['memberships']),
+    ...mapGetters('tenant', ['tenant', 'tenants'])
+  },
+  methods: {
+    ...mapActions('tenant', ['getTenants', 'setCurrentTenant']),
+    role(item) {
+      let role
+      switch (item) {
+        case 'USER':
+          role = 'User'
+          break
+        case 'READ_ONLY_USER':
+          role = 'Restricted User'
+          break
+        case 'TENANT_ADMIN':
+          role = 'Administrator'
+          break
+        default:
+          role = ''
+          break
+      }
+      return role
+    },
+    async removeUser(removalTenant) {
+      this.isRemovingUser = true
+      const initialTenant = this.tenant
+
+      let membershipsObject = {}
+      for (const membership of this.memberships) {
+        membershipsObject[membership.tenant.id] = membership.id
+      }
+      const membershipId = membershipsObject[removalTenant.id]
+
+      if (initialTenant.id !== removalTenant.id) {
+        await this.setCurrentTenant(removalTenant.slug)
+      }
+
+      const res = await this.$apollo.mutate({
+        mutation: require('@/graphql/Tenant/delete-membership.gql'),
+        variables: { membershipId }
+      })
+
+      if (res?.data?.delete_membership?.success) {
+        this.$emit(
+          'successful-action',
+          'You have removed yourself from this team.'
+        )
+      } else {
+        this.$emit(
+          'failed-action',
+          'Something went wrong while trying to leave this team. Please try again.'
+        )
+      }
+
+      if (initialTenant.id !== removalTenant.id) {
+        await this.setCurrentTenant(initialTenant.slug)
+      } else if (this.tenants[0].id === removalTenant.id) {
+        await this.setCurrentTenant(this.tenants[1].slug)
+      } else {
+        await this.setCurrentTenant(this.tenants[0].slug)
+      }
```

Comment:
I don't love this, but needed to figure out a way to switch to a tenant you're still a part of. Happy to hear suggestions!